### PR TITLE
PPPLMER-106170: changes to update checkout button role as link as per suggested by accessibility

### PR DESCRIPTION
**Pull request by mpreddy928**
### Description

.There is one production accessibility issue from Paypal detected by our ADA partner, and it’s a level 1 priority ticket which means we need to fix it in order to get the conformance letter. Could you help to see if anyone from your tech POC can help with it? What is the issue? Actual result The element "Paypal", link is not coded as link. Expected result The screen reader announces the element "Paypal" with the correct role "link". Steps to reproduce 1. Go to the URL. 2. Navigate to the element "Paypal" present under heading "Order summary". 3. Inspect the element "Paypal". 4. Notice that the element's role does not match its function. Recommendation Use role="link" for the element "PayPal". Resources Using ARIA: Roles, states, and properties.

Reported website -  http://www.tumi.com/

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues
Requirement from https://engineering.paypalcorp.com/jira/browse/PPPLMER-106170
and suggestions from ask-accessibility team

### Reproduction Steps (if applicable)
Open paypal checkout button in any merchant website inspect to the Payapl button its using  role="button" for parent div and suggested to add role="link" to support accessibility screen reader.

### Screenshots (if applicable)
Screen Shot 2022-03-30 at 2 46 38 PM (1)

### Dependent Changes (if applicable)

❤️  Thank you!

### CODE REVIEW

File: src/ui/buttons/button.jsx [hunk 1]
```diff
@@ -215,7 +215,7 @@ export function Button({ fundingSource, style, multiple, locale, env, fundingEli
             ].join(' ') }
         >
             <div
-                role='button'
+                role='link'
```
Comment (gregjopa) [inline src/ui/buttons/button.jsx:218]: @paypal/checkout-sdk please review and share any concerns. This change should improve screen reader behavior.

Comment (mpreddy928) [reply src/ui/buttons/button.jsx:218]: For more info on the reference for the discussion with accessibility team here is the reference - https://paypal.slack.com/archives/C1S7705Q8/p1648757897237639?thread_ts=1648751542.872819&cid=C1S7705Q8

**Review by wsbrunson — APPROVED** [review]

**Comment by mpreddy928** [comment]
@wsbrunson - can i know when this PR can be merged as we need to provide ETA in the JIRA? i don't merge option my side

**Comment by wsbrunson** [comment]
we can merge this today and it will be ready to go for the js-sdk deploy on Wednesday, 20 April

**Pull request merged**